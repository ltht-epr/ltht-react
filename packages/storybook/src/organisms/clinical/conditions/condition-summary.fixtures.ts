import {
  Metadata,
  Condition,
  ConditionClinicalStatus,
  ConditionVerificationStatus,
  PartialDateTimeKindCode,
} from '@ltht-react/types'

const mockMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
  isRedacted: false,
  requestedWhen: '',
}

const redactedMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
  isRedacted: true,
  requestedWhen: '',
}

const ConditionOne: Condition = {
  id: '017ca927-67e0-e581-a2c7-115087226fg2|condition',
  metadata: redactedMetadata,
  category: [
    {
      coding: [
        {
          code: '55607006',
          display: 'Problem',
        },
      ],
      text: 'Problem',
    },
  ],
  code: {
    coding: [
      {
        system: 'HEALTHPLUG',
        code: '60123008',
        display: 'Delusional disorder, mixed type',
      },
    ],
    text: 'Delusional disorder, mixed type',
  },
  clinicalStatus: ConditionClinicalStatus.Active,
  verificationStatus: ConditionVerificationStatus.Confirmed,
  assertedDate: {
    value: '2020-02-01T00:00:00+00:00',
    kind: PartialDateTimeKindCode.Date,
  },
  severity: {
    coding: [{ code: '2448400', display: 'Severe' }],
    text: 'Severe',
  },
}

const ConditionTwo: Condition = {
  id: '345ha927-34e0-e911-a2c7-005087226er3|condition',
  metadata: mockMetadata,
  category: [
    {
      coding: [
        {
          code: '55607006',
          display: 'Problem',
        },
      ],
      text: 'Problem',
    },
  ],
  code: {
    coding: [
      {
        system: 'HEALTHPLUG',
        code: '44054006',
        display: 'Diabetes mellitus type 2',
      },
    ],
    text: 'Diabetes mellitus type 2',
  },
  clinicalStatus: ConditionClinicalStatus.Remission,
  assertedDate: {
    value: '2020-01-01T00:00:00+00:00',
    kind: PartialDateTimeKindCode.Date,
  },
  verificationStatus: ConditionVerificationStatus.Refuted,
  severity: {
    coding: [{ code: '2448402', display: 'Mild' }],
    text: 'Mild',
  },
}

const conditions: Condition[] = [ConditionOne, ConditionTwo]

export default conditions
