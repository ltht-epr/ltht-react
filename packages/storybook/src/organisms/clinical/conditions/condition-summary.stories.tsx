import React from 'react'
import { storiesOf } from '@storybook/react'
import JSXAddon from 'storybook-addon-jsx'

import ConditionSummary from '@ltht-react/condition-summary'
import readme from '@ltht-react/condition-summary/README.md'
import { Card, CardHeader, CardList, CardBody } from '@ltht-react/card'
import conditions from './condition-summary.fixtures'

const stories = storiesOf('Organisms - Clinical|Condition', module) as any

stories.addWithJSX = JSXAddon.addWithJSX

stories.addParameters({
  readme: {
    sidebar: readme,
  },
})

stories.addWithJSX('Summary', () => {
  return (
    <Card noData={!conditions}>
      <CardHeader>
        <h3>Conditions</h3>
      </CardHeader>
      <CardBody>
        <CardList>
          <ConditionSummary conditions={conditions} />
        </CardList>
      </CardBody>
    </Card>
  )
})
