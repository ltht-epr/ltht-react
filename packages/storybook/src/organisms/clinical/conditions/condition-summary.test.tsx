import React from 'react'
import { mount } from 'enzyme'
import ReactDOM from 'react-dom'

import ConditionSummary from '@ltht-react/condition-summary'
import conditions from './condition-summary.fixtures'

describe('ConditionSummary', () => {
  it('renders without crashing', () => {
    const ul = document.createElement('ul')
    ReactDOM.render(<ConditionSummary conditions={conditions} />, ul)
  })
  it('matches snapshot', () => {
    expect(mount(<ConditionSummary conditions={conditions} />)).toMatchSnapshot('wrapper mount')
  })
})
