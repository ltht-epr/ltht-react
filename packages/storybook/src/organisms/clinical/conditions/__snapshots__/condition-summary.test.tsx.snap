// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConditionSummary matches snapshot: wrapper mount 1`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: left;
}

.emotion-0 {
  color: #768692;
}

.emotion-3 {
  text-align: right;
}

.emotion-2 {
  color: #231f20;
}

.emotion-7 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-5 {
  color: #231f20;
  text-align: left;
}

.emotion-6 {
  color: #768692;
  text-align: left;
  font-size: smaller;
  padding-top: 0.25rem;
}

.emotion-10 {
  text-align: right;
}

.emotion-9 {
  color: #768692;
  text-align: right;
  font-size: smaller;
  padding-top: 0.25rem;
}

<ConditionSummary
  conditions={
    Array [
      Object {
        "assertedDate": Object {
          "kind": "DATE",
          "value": "2020-02-01T00:00:00+00:00",
        },
        "category": Array [
          Object {
            "coding": Array [
              Object {
                "code": "55607006",
                "display": "Problem",
              },
            ],
            "text": "Problem",
          },
        ],
        "clinicalStatus": "ACTIVE",
        "code": Object {
          "coding": Array [
            Object {
              "code": "60123008",
              "display": "Delusional disorder, mixed type",
              "system": "HEALTHPLUG",
            },
          ],
          "text": "Delusional disorder, mixed type",
        },
        "id": "017ca927-67e0-e581-a2c7-115087226fg2|condition",
        "metadata": Object {
          "dataSources": Array [
            Object {
              "display": "Mock",
            },
          ],
          "isRedacted": true,
          "requestedWhen": "",
        },
        "severity": Object {
          "coding": Array [
            Object {
              "code": "2448400",
              "display": "Severe",
            },
          ],
          "text": "Severe",
        },
        "subject": Object {
          "display": "Mr Peter Humble",
          "typeName": "test",
        },
        "verificationStatus": "CONFIRMED",
      },
      Object {
        "assertedDate": Object {
          "kind": "DATE",
          "value": "2020-01-01T00:00:00+00:00",
        },
        "category": Array [
          Object {
            "coding": Array [
              Object {
                "code": "55607006",
                "display": "Problem",
              },
            ],
            "text": "Problem",
          },
        ],
        "clinicalStatus": "REMISSION",
        "code": Object {
          "coding": Array [
            Object {
              "code": "44054006",
              "display": "Diabetes mellitus type 2",
              "system": "HEALTHPLUG",
            },
          ],
          "text": "Diabetes mellitus type 2",
        },
        "id": "345ha927-34e0-e911-a2c7-005087226er3|condition",
        "metadata": Object {
          "dataSources": Array [
            Object {
              "display": "Mock",
            },
          ],
          "isRedacted": false,
          "requestedWhen": "",
        },
        "severity": Object {
          "coding": Array [
            Object {
              "code": "2448402",
              "display": "Mild",
            },
          ],
          "text": "Mild",
        },
        "subject": Object {
          "display": "Mr Joe BLoggs",
          "typeName": "test",
        },
        "verificationStatus": "REFUTED",
      },
    ]
  }
>
  <ListItem
    key="017ca927-67e0-e581-a2c7-115087226fg2|condition"
  >
    <li>
      <ConditionSummaryItem
        condition={
          Object {
            "assertedDate": Object {
              "kind": "DATE",
              "value": "2020-02-01T00:00:00+00:00",
            },
            "category": Array [
              Object {
                "coding": Array [
                  Object {
                    "code": "55607006",
                    "display": "Problem",
                  },
                ],
                "text": "Problem",
              },
            ],
            "clinicalStatus": "ACTIVE",
            "code": Object {
              "coding": Array [
                Object {
                  "code": "60123008",
                  "display": "Delusional disorder, mixed type",
                  "system": "HEALTHPLUG",
                },
              ],
              "text": "Delusional disorder, mixed type",
            },
            "id": "017ca927-67e0-e581-a2c7-115087226fg2|condition",
            "metadata": Object {
              "dataSources": Array [
                Object {
                  "display": "Mock",
                },
              ],
              "isRedacted": true,
              "requestedWhen": "",
            },
            "severity": Object {
              "coding": Array [
                Object {
                  "code": "2448400",
                  "display": "Severe",
                },
              ],
              "text": "Severe",
            },
            "subject": Object {
              "display": "Mr Peter Humble",
              "typeName": "test",
            },
            "verificationStatus": "CONFIRMED",
          }
        }
      >
        <div
          className="emotion-4"
          role="link"
        >
          <ConditionRedacted
            condition={
              Object {
                "assertedDate": Object {
                  "kind": "DATE",
                  "value": "2020-02-01T00:00:00+00:00",
                },
                "category": Array [
                  Object {
                    "coding": Array [
                      Object {
                        "code": "55607006",
                        "display": "Problem",
                      },
                    ],
                    "text": "Problem",
                  },
                ],
                "clinicalStatus": "ACTIVE",
                "code": Object {
                  "coding": Array [
                    Object {
                      "code": "60123008",
                      "display": "Delusional disorder, mixed type",
                      "system": "HEALTHPLUG",
                    },
                  ],
                  "text": "Delusional disorder, mixed type",
                },
                "id": "017ca927-67e0-e581-a2c7-115087226fg2|condition",
                "metadata": Object {
                  "dataSources": Array [
                    Object {
                      "display": "Mock",
                    },
                  ],
                  "isRedacted": true,
                  "requestedWhen": "",
                },
                "severity": Object {
                  "coding": Array [
                    Object {
                      "code": "2448400",
                      "display": "Severe",
                    },
                  ],
                  "text": "Severe",
                },
                "subject": Object {
                  "display": "Mr Peter Humble",
                  "typeName": "test",
                },
                "verificationStatus": "CONFIRMED",
              }
            }
          >
            <div
              className="emotion-1"
            >
              <RedactedDescription>
                <div
                  className="emotion-0"
                >
                  Insufficient privileges
                </div>
              </RedactedDescription>
            </div>
            <div
              className="emotion-3"
            >
              <DateSummary
                datetime={
                  Object {
                    "kind": "DATE",
                    "value": "2020-02-01T00:00:00+00:00",
                  }
                }
              >
                <div
                  className="emotion-2"
                >
                  Feb-01,-2020
                </div>
              </DateSummary>
            </div>
          </ConditionRedacted>
        </div>
      </ConditionSummaryItem>
    </li>
  </ListItem>
  <ListItem
    key="345ha927-34e0-e911-a2c7-005087226er3|condition"
  >
    <li>
      <ConditionSummaryItem
        condition={
          Object {
            "assertedDate": Object {
              "kind": "DATE",
              "value": "2020-01-01T00:00:00+00:00",
            },
            "category": Array [
              Object {
                "coding": Array [
                  Object {
                    "code": "55607006",
                    "display": "Problem",
                  },
                ],
                "text": "Problem",
              },
            ],
            "clinicalStatus": "REMISSION",
            "code": Object {
              "coding": Array [
                Object {
                  "code": "44054006",
                  "display": "Diabetes mellitus type 2",
                  "system": "HEALTHPLUG",
                },
              ],
              "text": "Diabetes mellitus type 2",
            },
            "id": "345ha927-34e0-e911-a2c7-005087226er3|condition",
            "metadata": Object {
              "dataSources": Array [
                Object {
                  "display": "Mock",
                },
              ],
              "isRedacted": false,
              "requestedWhen": "",
            },
            "severity": Object {
              "coding": Array [
                Object {
                  "code": "2448402",
                  "display": "Mild",
                },
              ],
              "text": "Mild",
            },
            "subject": Object {
              "display": "Mr Joe BLoggs",
              "typeName": "test",
            },
            "verificationStatus": "REFUTED",
          }
        }
      >
        <div
          className="emotion-4"
          role="link"
        >
          <div
            className="emotion-7"
          >
            <ConditionTitle
              condition={
                Object {
                  "assertedDate": Object {
                    "kind": "DATE",
                    "value": "2020-01-01T00:00:00+00:00",
                  },
                  "category": Array [
                    Object {
                      "coding": Array [
                        Object {
                          "code": "55607006",
                          "display": "Problem",
                        },
                      ],
                      "text": "Problem",
                    },
                  ],
                  "clinicalStatus": "REMISSION",
                  "code": Object {
                    "coding": Array [
                      Object {
                        "code": "44054006",
                        "display": "Diabetes mellitus type 2",
                        "system": "HEALTHPLUG",
                      },
                    ],
                    "text": "Diabetes mellitus type 2",
                  },
                  "id": "345ha927-34e0-e911-a2c7-005087226er3|condition",
                  "metadata": Object {
                    "dataSources": Array [
                      Object {
                        "display": "Mock",
                      },
                    ],
                    "isRedacted": false,
                    "requestedWhen": "",
                  },
                  "severity": Object {
                    "coding": Array [
                      Object {
                        "code": "2448402",
                        "display": "Mild",
                      },
                    ],
                    "text": "Mild",
                  },
                  "subject": Object {
                    "display": "Mr Joe BLoggs",
                    "typeName": "test",
                  },
                  "verificationStatus": "REFUTED",
                }
              }
            >
              <div
                className="emotion-5"
              >
                Diabetes mellitus type 2
              </div>
            </ConditionTitle>
            <ConditionCategory
              condition={
                Object {
                  "assertedDate": Object {
                    "kind": "DATE",
                    "value": "2020-01-01T00:00:00+00:00",
                  },
                  "category": Array [
                    Object {
                      "coding": Array [
                        Object {
                          "code": "55607006",
                          "display": "Problem",
                        },
                      ],
                      "text": "Problem",
                    },
                  ],
                  "clinicalStatus": "REMISSION",
                  "code": Object {
                    "coding": Array [
                      Object {
                        "code": "44054006",
                        "display": "Diabetes mellitus type 2",
                        "system": "HEALTHPLUG",
                      },
                    ],
                    "text": "Diabetes mellitus type 2",
                  },
                  "id": "345ha927-34e0-e911-a2c7-005087226er3|condition",
                  "metadata": Object {
                    "dataSources": Array [
                      Object {
                        "display": "Mock",
                      },
                    ],
                    "isRedacted": false,
                    "requestedWhen": "",
                  },
                  "severity": Object {
                    "coding": Array [
                      Object {
                        "code": "2448402",
                        "display": "Mild",
                      },
                    ],
                    "text": "Mild",
                  },
                  "subject": Object {
                    "display": "Mr Joe BLoggs",
                    "typeName": "test",
                  },
                  "verificationStatus": "REFUTED",
                }
              }
            >
              <div
                className="emotion-6"
              >
                Problem - Mild
              </div>
            </ConditionCategory>
          </div>
          <div
            className="emotion-10"
          >
            <DateSummary
              datetime={
                Object {
                  "kind": "DATE",
                  "value": "2020-01-01T00:00:00+00:00",
                }
              }
            >
              <div
                className="emotion-2"
              >
                Jan-01,-2020
              </div>
            </DateSummary>
            <ConditionStatus
              condition={
                Object {
                  "assertedDate": Object {
                    "kind": "DATE",
                    "value": "2020-01-01T00:00:00+00:00",
                  },
                  "category": Array [
                    Object {
                      "coding": Array [
                        Object {
                          "code": "55607006",
                          "display": "Problem",
                        },
                      ],
                      "text": "Problem",
                    },
                  ],
                  "clinicalStatus": "REMISSION",
                  "code": Object {
                    "coding": Array [
                      Object {
                        "code": "44054006",
                        "display": "Diabetes mellitus type 2",
                        "system": "HEALTHPLUG",
                      },
                    ],
                    "text": "Diabetes mellitus type 2",
                  },
                  "id": "345ha927-34e0-e911-a2c7-005087226er3|condition",
                  "metadata": Object {
                    "dataSources": Array [
                      Object {
                        "display": "Mock",
                      },
                    ],
                    "isRedacted": false,
                    "requestedWhen": "",
                  },
                  "severity": Object {
                    "coding": Array [
                      Object {
                        "code": "2448402",
                        "display": "Mild",
                      },
                    ],
                    "text": "Mild",
                  },
                  "subject": Object {
                    "display": "Mr Joe BLoggs",
                    "typeName": "test",
                  },
                  "verificationStatus": "REFUTED",
                }
              }
            >
              <div
                className="emotion-9"
              >
                Remission - Refuted
              </div>
            </ConditionStatus>
          </div>
        </div>
      </ConditionSummaryItem>
    </li>
  </ListItem>
</ConditionSummary>
`;
